cfme/services/catalogs/catalog_item.py

```diff
@@ -152,7 +152,8 @@ def create(self):
                                'select_catalog': catalog.name,
                                'select_dialog': dialog.name,
                                'select_orch_template': self.orch_template,
-                               'select_provider': self.provider,
+                               'select_provider': self.provider if self.provider.type_tclass
```

Comment:
This line it a little messy, the way it's broken up, let's assign what should go in there to a variable on line 148 to make it cleaner.